repos/system_upgrade/common/actors/updategrubcore/libraries/updategrubcore.py

```diff
@@ -3,33 +3,39 @@
 from leapp.libraries.stdlib import api, CalledProcessError, config, run
 
 
-def update_grub_core(grub_dev):
+def update_grub_core(grub_devs):
     """
     Update GRUB core after upgrade from RHEL7 to RHEL8
 
     On legacy systems, GRUB core does not get automatically updated when GRUB packages
     are updated.
     """
-    cmd = ['grub2-install', grub_dev]
-    if config.is_debug():
-        cmd += ['-v']
-    try:
-        run(cmd)
-    except CalledProcessError as err:
-        reporting.create_report([
-            reporting.Title('GRUB core update failed'),
-            reporting.Summary(str(err)),
-            reporting.Groups([reporting.Groups.BOOT]),
-            reporting.Severity(reporting.Severity.HIGH),
-            reporting.Remediation(
-                hint='Please run "grub2-install <GRUB_DEVICE>" manually after upgrade'
-            )
-        ])
-        api.current_logger().warning('GRUB core update on {} failed'.format(grub_dev))
-        raise StopActorExecution()
+
+    successful = []
+    for dev in grub_devs:
+        cmd = ['grub2-install', dev]
+        if config.is_debug():
+            cmd += ['-v']
+        try:
+            run(cmd)
+        except CalledProcessError as err:
+            reporting.create_report([
+                reporting.Title('GRUB core update failed'),
+                reporting.Summary(str(err)),
+                reporting.Groups([reporting.Groups.BOOT]),
+                reporting.Severity(reporting.Severity.HIGH),
+                reporting.Remediation(
+                    hint='Please run "grub2-install <GRUB_DEVICE>" manually after upgrade'
+                )
+            ])
+            api.current_logger().warning('GRUB core update on {} failed'.format(dev))
+            raise StopActorExecution()
```

Comment:
Same question about passing details to StopActorExecution exception as in mdraid library

Reply:
StopActorExecution != StopActorExecutionError, the former just terminates the actor, allowing the leapp execution to continue. I believe you can't even add the message there.

The same goes for the other places you mentioned.

Reply:
Yeah, you are right, it stops without reporting anything https://github.com/oamg/leapp/blob/master/leapp/actors/__init__.py#L290
Sorry, it was an early Friday review, disregard my comments then!